src/components/app/data/utils.js

```diff
@@ -83,29 +110,28 @@ export function determineLearnerHasContentAssignmentsOnly({
   redeemableLearnerCreditPolicies,
   hasCurrentEnterpriseOffers,
 }) {
-  const hasActiveLicense = !!(subscriptionPlan?.isActive && subscriptionLicense?.status === LICENSE_STATUS.ACTIVATED);
-  const hasActiveLicenseOrLicenseRequest = hasActiveLicense || licenseRequests.length > 0;
-
-  const hasAssignedCodesOrCodeRequests = couponCodesCount > 0 || couponCodeRequests.length > 0;
-  const autoAppliedPolicyTypes = [
-    POLICY_TYPES.PER_LEARNER_CREDIT,
-    POLICY_TYPES.PER_ENROLLMENT_CREDIT,
-  ];
-  const hasAutoAppliedLearnerCreditPolicies = !!redeemableLearnerCreditPolicies.redeemablePolicies.filter(
-    policy => autoAppliedPolicyTypes.includes(policy.policyType),
-  ).length > 0;
-  const hasAllocatedOrAcceptedAssignments = !!(
-    redeemableLearnerCreditPolicies.learnerContentAssignments.hasAllocatedAssignments
-    || redeemableLearnerCreditPolicies.learnerContentAssignments.hasAcceptedAssignments
-  );
+  const hasAssignments = hasAllocatedOrAcceptedAssignments({
+    redeemableLearnerCreditPolicies,
+  });
+  // If the enterprise learner does not have any applicable assignments, we can return early to avoid additional checks
+  if (!hasAssignments) {
+    return false;
+  }
 
-  return (
-    hasAllocatedOrAcceptedAssignments
-    && !hasCurrentEnterpriseOffers
-    && !hasActiveLicenseOrLicenseRequest
-    && !hasAssignedCodesOrCodeRequests
-    && !hasAutoAppliedLearnerCreditPolicies
-  );
+  // We check for any true values for the following cases. If any cases return true, this indicates that the
+  // enterprise learner is not an assignment only learner. We default return to true when no other subsidy exist
+  switch (true) {
+    case hasCurrentEnterpriseOffers:
+      return false;
+    case hasActiveLicenseOrLicenseRequest({ subscriptionPlan, subscriptionLicense, licenseRequests }):
+      return false;
+    case hasAutoAppliedLearnerCreditPolicies({ redeemableLearnerCreditPolicies }):
+      return false;
+    case hasAssignedCodesOrCodeRequests({ couponCodeRequests, couponCodesCount }):
+      return false;
+    default:
+      return true;
```

Comment:
nit: might consider adding a comment above this line to more explicitly denote that `true` here reflects an assignment only learner.